Snabble/UI/Scanner/CheckoutBar.swift

```diff
@@ -29,44 +30,147 @@ final class CheckoutBar: NibView {
     weak var shoppingCartDelegate: ShoppingCartDelegate?
 
     init(_ parentVC: UIViewController & AnalyticsDelegate, _ shoppingCart: ShoppingCart) {
+
         self.parentVC = parentVC
         self.shoppingCart = shoppingCart
-
         super.init(frame: .zero)
+
+        self.setupUI()
+
+        if let methodSelectionStackView = self.methodSelectionStackView, let methodIcon = self.methodIcon {
+            self.methodSelector = PaymentMethodSelector(parentVC, methodSelectionStackView, methodIcon, self.shoppingCart)
+        }
+        self.methodSelector?.delegate = self
+
     }
 
     public required init?(coder aDecoder: NSCoder) {
         fatalError("init(coder:) has not been implemented")
     }
 
-    override func awakeFromNib() {
-        super.awakeFromNib()
-
-        self.checkoutButton.addTarget(self, action: #selector(checkoutTapped(_:)), for: .touchUpInside)
-        self.checkoutButton.setTitle(L10n.Snabble.Shoppingcart.BuyProducts.now, for: .normal)
-        self.checkoutButton.makeSnabbleButton()
-
+    private func setupUI() {
+
+        let contentStack = UIStackView()
+        contentStack.translatesAutoresizingMaskIntoConstraints = false
+        contentStack.axis = .vertical
+        contentStack.distribution = .fill
+        contentStack.alignment = .fill
+        contentStack.spacing = 0
+
+        let summaryView = UIView()
+        summaryView.translatesAutoresizingMaskIntoConstraints = false
```

Comment:
I would recommend to use an `UILayoutGuide` instead of an `UIView` just for positioning inner elements.